Exception/AbstractAlgoliaProductException.php

```diff
@@ -0,0 +1,71 @@
+<?php
+/**
+ * Module Algolia Algoliasearch
+ */
+namespace Algolia\AlgoliaSearch\Exception;
+
+use Magento\Catalog\Model\Product;
+
+/**
+ * Class: AbstractAlgoliaProductException
+ */
+abstract class AbstractAlgoliaProductException extends \Exception
```

Comment:
Could you rename it to `AlgoliaProductReindexingException`? The word `Abstract` is redundant and adding `Reindexing` it's more descriptive and understandable.

This kind of exception can happen only in runtime (and controls the runtime) and in this case it's better to extend [`\RuntimeException`](http://php.net/manual/en/class.runtimeexception.php) instead of generic `\Exception`.